internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -577,6 +570,50 @@ func (srv *Server) Serve(Port int) {
 		})
 	})
 
+	r.Route("/ws", func(r chi.Router) {
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			m.HandleRequest(w, r)
+		})
+	})
+
+	m.HandleMessage(func(s *melody.Session, msg []byte) {
+		log.Println(string(msg))
+		params := strings.Split(string(msg), " ")
+		lock.Lock()
+		if params[0] == "edit" {
+			if currentlyViewing[params[1]] {
+				_, exists := s.Get(params[1])
+				if !exists {
+					log.Println("ATTEMPT TO EDIT RULE THAT IS BEING EDITED")
+					s.Write([]byte("INUSE"))
```

Comment:
I'm guessing these are going to be read by the browser client right? You're probably going to want some sort of message delimiter, otherwise they won't know when the message has ended. Something simple like a newline works for this case.

Reply:
Hmm, do you mean in the case of like a network outage or like packet corruption? I was intending for the message that is getting emitted back to only ever be a couple of options, but I agree that there should be some sort of delimiter to explicitly state when the message is terminated

Reply:
Ah sorry, didn't realize that web socket protocol handles its own framing and stuff. So you're good :)